montepy/data_inputs/material.py

```diff
@@ -102,13 +105,23 @@ def is_atom_fraction(self):
     @property
     def material_components(self):
         """
-        The internal dictionary containing all the components of this material.
+            The internal dictionary containing all the components of this material.
 
-        The keys are :class:`~montepy.data_inputs.isotope.Isotope` instances, and the values are
-        :class:`~montepy.data_inputs.material_component.MaterialComponent` instances.
+        .. deprecated:: 0.4.1
+            MaterialComponent has been deprecated as part of a redesign for the material
+            interface due to a critical bug in how MontePy handles duplicate nuclides.
+            See :ref:`migrate 0 1`.
 
-        :rtype: dict
+            The keys are :class:`~montepy.data_inputs.isotope.Isotope` instances, and the values are
+            :class:`~montepy.data_inputs.material_component.MaterialComponent` instances.
+
+            :rtype: dict
         """
+        warnings.warn(
+            f"""material_components is deprecated, and will be removed in MontePy 1.0.0.
+See <https://www.montepy.org/migrate0_1.html> for more information """,
+            DeprecationWarning,
```

Comment:
https://stackoverflow.com/q/55377831

I think `DeprecationWarning` is correct

Reply:
I did think about it a little bit. Given this discussion: I don't think there's enough down stream devs that they would see this. If anything I feel like `FutureWarning` might be the safest bet?